### PR TITLE
HPCC-18523 Ensure multi-value properties have correct binary flag

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

There were no proper unittests for IPT.
I've added some and used them to test this (and expose a couple of other issues), but will release in a separate issue/PR.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: system/jlib/jptree.cpp [hunk 1]
```diff
@@ -2572,6 +2572,10 @@ void PTree::addLocal(size32_t l, const void *data, bool _binary, int pos)
     IPTArrayValue *newValue = new CPTValue(l, data, _binary);
     Owned<IPropertyTree> tree = create(queryName(), newValue);
     PTree *_tree = QUERYINTERFACE(tree.get(), PTree); assertex(_tree); _tree->setParent(this);
+    if (_binary)
+        _tree->flags |= ipt_binary;
+    else
+        _tree->flags &= ~ipt_binary;
```
Comment (mckellyln) [inline system/jlib/jptree.cpp]: Why not, just for consistency, use IptFlagSet(_tree->flags, ipt_binary) here ?

**Review by mckellyln — APPROVED** [review]
Looks ok, one minor comment

File: system/jlib/jptree.cpp (hunk 1 above)
Comment (jakesmith) [reply system/jlib/jptree.cpp]: You're right, I should. I will change.

Comment (jakesmith) [reply system/jlib/jptree.cpp]: changed

**Comment by jakesmith** [comment]
@mckellyln - addressed comment.

**Pull request merged**